server/src/main/java/io/deephaven/server/barrage/BarrageStreamGenerator.java

```diff
@@ -797,6 +799,60 @@ private ByteBuffer getSubscriptionMetadata(final SubView view) throws IOExceptio
         return header.dataBuffer().slice();
     }
 
+    private ByteBuffer getSnapshotMetadata(final SnapshotView view) throws IOException {
+        final FlatBufferBuilder metadata = new FlatBufferBuilder();
+
+        int effectiveViewportOffset = 0;
+        if (view.isViewport()) {
+            try (final RowSetGenerator viewportGen = new RowSetGenerator(view.viewport)) {
+                effectiveViewportOffset = viewportGen.addToFlatBuffer(metadata);
+            }
+        }
+
+        int effectiveColumnSetOffset = 0;
+        if (isSnapshot && view.subscribedColumns != null) {
+            effectiveColumnSetOffset = new BitSetGenerator(view.subscribedColumns).addToFlatBuffer(metadata);
+        }
+
+        final int rowsAddedOffset = rowsAdded.addToFlatBuffer(metadata);
+        final int rowsRemovedOffset = 0;
+
+        final int shiftDataOffset = shifted.addToFlatBuffer(metadata);
+
+        // Added Chunk Data:
+        int addedRowsIncludedOffset = rowsIncluded.addToFlatBuffer(metadata);
+
+        final int nodesOffset = 0;
+
+        BarrageUpdateMetadata.startBarrageUpdateMetadata(metadata);
+        BarrageUpdateMetadata.addNumAddBatches(metadata,
+                LongSizedDataStructure.intSize("BarrageStreamGenerator", view.numAddBatches));
+        BarrageUpdateMetadata.addNumModBatches(metadata,
+                LongSizedDataStructure.intSize("BarrageStreamGenerator", 0));
+        BarrageUpdateMetadata.addIsSnapshot(metadata, isSnapshot);
```

Comment:
I think there's something to be said for hard-coding constants that can never be changed.